tools/testrunner/testrunner.go

```diff
@@ -76,61 +91,36 @@ func newRunner(gotestsumExecutable string) *runner {
 
 func (r *runner) sanitizeAndParseArgs(args []string) ([]string, error) {
 	var sanitizedArgs []string
-	type action struct {
-		f   func(string) error
-		err string
-	}
-	var next *action
-	for i, arg := range args {
-		if next != nil {
-			if err := next.f(arg); err != nil {
-				return nil, err
-			}
-			next = nil
-			continue
-		} else if arg == "-retries" {
-			next = &action{
-				f: func(arg string) error {
-					var err error
-					r.retries, err = strconv.Atoi(arg)
-					return err
-				},
-				err: "got -retries flag with no value",
-			}
-			continue
-		} else if strings.HasPrefix(arg, "-retries=") {
+	for _, arg := range args {
+		if strings.HasPrefix(arg, retriesFlag) {
 			var err error
-			r.retries, err = strconv.Atoi(arg[len("-retries="):])
+			r.retries, err = strconv.Atoi(strings.Split(arg, "=")[1])
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("invalid argument %q: %w", retriesFlag, err)
 			}
-			continue
-		} else if arg == "--junitfile" {
-			// --junitfile is used by gotestsum
-			next = &action{
-				f: func(arg string) error {
-					r.junitOutputPath = arg
-					return nil
-				},
-				err: "got --junitfile flag with no value",
+			if r.retries == 0 {
+				return nil, fmt.Errorf("invalid argument %q: must be greater than zero", retriesFlag)
 			}
 			continue
-		} else if strings.HasPrefix(arg, "--junitfile=") {
+		} else if strings.HasPrefix(arg, coverProfileFlag) {
+			r.coverProfilePath = strings.Split(arg, "=")[1]
+			continue
+		} else if strings.HasPrefix(arg, junitReportFlag) {
 			// --junitfile is used by gotestsum
-			r.junitOutputPath = arg[len("--junitfile="):]
+			r.junitOutputPath = strings.Split(arg, "=")[1]
 			continue
-		} else if arg == "--" {
-			// Forward all arguments from -- on.
-			sanitizedArgs = append(sanitizedArgs, args[i:]...)
-			break
 		}
+		// Pass through all other arguments.
 		sanitizedArgs = append(sanitizedArgs, arg)
 	}
-	if next != nil {
-		return nil, fmt.Errorf("incomplete command line arguments: %s", next.err)
-	}
 	if r.junitOutputPath == "" {
-		return nil, fmt.Errorf("missing required argument --junitfile")
+		return nil, fmt.Errorf("missing required argument %q", junitReportFlag)
+	}
+	if r.coverProfilePath == "" {
+		return nil, fmt.Errorf("missing required argument %q", coverProfileFlag)
+	}
+	if r.retries == 0 {
+		return nil, fmt.Errorf("missing required argument %q", retriesFlag)
```

Comment:
Made this a required argument since there's really no point to having it be missing/zero.